s/syntax.ss

```diff
@@ -4692,6 +4692,17 @@
         [_ (syntax-error name "invalid library name")])))
 
   (define library-search
+    (lambda (who path dir* all-ext*)
+      (let-values ([(src-path obj-path obj-exists?) ((library-search-handler) who path dir* all-ext*)])
+        (unless (or (not src-path) (string? src-path))
+          ($oops 'library-search-handler "returned invalid source-file path ~s" src-path))
+        (unless (or (not obj-path) (string? obj-path))
+          ($oops 'library-search-handler "returned invalid object-file path ~s" obj-path))
+        (when (and obj-exists? (not obj-path))
+          ($oops 'library-search-handler "claimed object file was found but returned no object-file path"))
+        (values src-path obj-path obj-exists?))))
+
+  (define default-library-search
```

Comment:
I think it might be nice to expose the `default-library-search` as a prim in the same way `library-search-handler` is exposed.  That way, if a programmer wants to avoid any other hooks that might have been put on the `library-search-handler` it is possible to get back to the base library search functionality.